test/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildADesktopExe.cs

```diff
@@ -33,12 +36,83 @@ public void It_builds_a_simple_desktop_app()
             });
         }
 
+        [WindowsOnlyTheory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void RuntimeIdentifiersInferredCorrectly(bool useRidGraph)
+        {
+            Func<string, string, string> findAssembly = (a, b) => default;
+            findAssembly = (string path, string file) =>
+            {
+                var dir = new DirectoryInfo(path);
+                foreach (var folder in dir.GetDirectories())
+                {
+                    var ret = findAssembly(folder.FullName, file);
+                    if (ret is not null)
+                    {
+                        return ret;
+                    }
+                }
+
+                foreach (var f in dir.GetFiles())
+                {
+                    if (Path.GetFileName(f.FullName).Equals(file, StringComparison.OrdinalIgnoreCase))
+                    {
+                        return f.FullName;
+                    }
+                }
+
+                return null;
+            };
+
+            var rootDirectory = Path.Combine(Directory.GetCurrentDirectory(), "..", "..", "..");
+            string microsoftNetBuildTasksAssembly = findAssembly(rootDirectory, "Microsoft.NET.Build.Tasks.dll");
+            string ridInferenceTargets = findAssembly(rootDirectory, "Microsoft.NET.RuntimeIdentifierInference.targets");
+            var projectContents = @$"
+<Project>
+  <PropertyGroup>
+    <TargetFramework>net472</TargetFramework>
+    <TargetFrameworkIdentifier>.NETFramework</TargetFrameworkIdentifier>
+    <UseRidGraph>{useRidGraph}</UseRidGraph>
+    <MicrosoftNETBuildTasksAssembly>{microsoftNetBuildTasksAssembly}</MicrosoftNETBuildTasksAssembly>
+    <HasRuntimeOutput>true</HasRuntimeOutput>
+  </PropertyGroup>
+
+  <Import Project=""{ridInferenceTargets}"" />
+</Project>";
+
+            var fileLocation = Path.GetTempFileName();
+
+            try
+            {
+                File.WriteAllText(fileLocation, projectContents);
+
+                var loggers = new List<ILogger>
+            {
+                new global::Microsoft.Build.Logging.ConsoleLogger(LoggerVerbosity.Detailed)
+            };
+
+                var collection = new ProjectCollection(null, loggers, ToolsetDefinitionLocations.Default);
```

Comment:
We shouldn't use MSBuild APIs like this to load a test project.  That mixes up stage 0 and stage 2.  Rather than redirect the in-process stage 0 MSBuild to use some pieces of stage 2, we should use stage 2 in a separate process to do the evaluation.  You can use the `GetValuesCommand` class for this.